hedera-node/src/test/java/com/hedera/services/legacy/unit/serialization/HFileMetaSerdeTest.java

```diff
@@ -0,0 +1,225 @@
+		// setup:
+		doStaticMocking();
+		byte[] pretend = "NOPE".getBytes();
+		// and:
+		DataOutputStream out = mock(DataOutputStream.class);
+		SerializableDataOutputStream serOut = mock(SerializableDataOutputStream.class);
+		// and:
+		ArgumentCaptor<JKeySerializer.StreamConsumer<DataOutputStream>> captor =
+				ArgumentCaptor.forClass(JKeySerializer.StreamConsumer.class);
+		InOrder inOrder = inOrder(out, serOut, serOutFactory, serdes);
+
+		given(streamContentDiscovery.discoverFor(captor.capture())).willReturn(pretend);
+		given(serOutFactory.apply(out)).willReturn(serOut);
+
+		// when:
+		var shouldBePretend = HFileMetaSerde.serialize(known);
+		// and:
+		var consumer = captor.getValue();
+		consumer.accept(out);
+
+		// then:
+		assertSame(pretend, shouldBePretend);
+		// and:
+		inOrder.verify(serOutFactory).apply(out);
+		inOrder.verify(serOut).writeLong(HFileMetaSerde.MEMO_VERSION);
+		inOrder.verify(serOut).writeBoolean(deleted);
+		inOrder.verify(serOut).writeLong(expiry);
+		inOrder.verify(serOut).writeNormalisedString(memo);
+		inOrder.verify(serdes).writeNullable(
+				argThat(wacl::equals),
+				argThat(serOut::equals),
+				any(IoWritingConsumer.class));
+
+		// cleanup:
+		undoStaticMocking();
+	}
+
+	@Test
+	public void legacySerdeTest() throws Exception {
+		// setup:
+		var fid = IdUtils.asFile("0.0.1001");
+
+		// given:
+		var expInfo = toGrpc(known, fid, 1024);
+		byte[] legacyRepr = Hex.decodeHex(hexedLegacyKnown);
+
+		// when:
+		var replica = deserialize(new DataInputStream(new ByteArrayInputStream(legacyRepr)));
+		// and:
+		var replicaInfo = toGrpc(replica, fid, 1024);
+
+		assertEquals(expInfo.getExpirationTime(), replicaInfo.getExpirationTime());
+		assertEquals(expInfo.getDeleted(), replicaInfo.getDeleted());
+		assertEquals(expInfo.getKeys().getKeysCount(), replicaInfo.getKeys().getKeysCount());
+	}
+
+	@SuppressWarnings("unchecked")
+	void doStaticMocking() {
+		serdes = mock(DomainSerdes.class);
+		discovery = mock(HFileMetaSerde.StreamContentDiscovery.class);
+		serInFactory = mock(Function.class);
+		serOutFactory = mock(Function.class);
+
+		serdes = mock(DomainSerdes.class);
+		HFileMetaSerde.serdes = serdes;
+		HFileMetaSerde.serInFactory = serInFactory;
+		HFileMetaSerde.serOutFactory = serOutFactory;
+		HFileMetaSerde.streamContentDiscovery = discovery;
+	}
+
+	void undoStaticMocking() {
+		HFileMetaSerde.serdes = new DomainSerdes();
+		HFileMetaSerde.serInFactory = SerializableDataInputStream::new;
+		HFileMetaSerde.serOutFactory = SerializableDataOutputStream::new;
+		HFileMetaSerde.streamContentDiscovery = JKeySerializer::byteStream;
+	}
+
+	public static HFileMeta convert(FileInfo fi) throws DecoderException {
+		return new HFileMeta(
+				fi.getDeleted(),
+				JKey.mapKey(Key.newBuilder().setKeyList(fi.getKeys()).build()),
+				fi.getExpirationTime().getSeconds());
+	}
+
+	public static FileInfo toGrpc(HFileMeta info, FileID fid, long size) throws Exception {
```

Comment:
can we move this out of test code as we use this static method in `FileServiceHandler`

Reply:
`FileServiceHandler` is under _src/test/java_ also :)...it's just kept around to keep running a handful of legacy unit tests.

Once the properties refactor is complete, I believe we can delete a huge amount of these random legacy classes.

Reply:
Ahh.. missed that part. Makes sense